aesara/scalar/math.py

```diff
@@ -1044,3 +1048,228 @@ def c_code(self, node, name, inp, out, sub):
 
 
 log1mexp = Log1mexp(upgrade_to_float, name="scalar_log1mexp")
+
+
+class BetaInc(TernaryScalarOp):
+    """
+    Regularized incomplete beta function
+    """
+
+    nfunc_spec = ("scipy.special.betainc", 3, 1)
+
+    def impl(self, a, b, x):
+        return scipy.special.betainc(a, b, x)
+
+    def grad(self, inp, grads):
+        a, b, x = inp
+        (gz,) = grads
+
+        return [
+            gz * betainc_dda_scalar(a, b, x),
+            gz * betainc_ddb_scalar(a, b, x),
+            gz
+            * exp(
+                log1p(-x) * (b - 1)
+                + log(x) * (a - 1)
+                - (gammaln(a) + gammaln(b) - gammaln(a + b))
+            ),
+        ]
+
+
+betainc = BetaInc(upgrade_to_float_no_complex, name="betainc")
+
+
+class BetaIncDda(TernaryScalarOp):
+    """
+    Gradient of the regularized incomplete beta function wrt to the first argument (a)
+    """
+
+    def impl(self, a, b, x):
+        return _betainc_derivative(a, b, x, wrtp=True)
+
+
+betainc_dda_scalar = BetaIncDda(upgrade_to_float_no_complex, name="betainc_dda")
+
+
+class BetaIncDdb(TernaryScalarOp):
+    """
+    Gradient of the regularized incomplete beta function wrt to the second argument (b)
+    """
+
+    def impl(self, a, b, x):
+        return _betainc_derivative(a, b, x, wrtp=False)
+
+
+betainc_ddb_scalar = BetaIncDdb(upgrade_to_float_no_complex, name="betainc_ddb")
```

Comment:
Before we merge this, let's combine these into a single `Op` with a boolean `wrtp` attribute.  The `_betainc_derivative` function can then become the entire `impl` method.

Reply:
Done, is this what you had in mind?

Reply:
If you refactor `_betainc_derivative` to be `BetaIncDdb.impl`, then yes.

Reply:
Got it